openlibrary/components/BulkSearch/utils/classes.js

```diff
@@ -122,10 +122,54 @@ export class AiExtractor extends AbstractExtractor{
     }
 }
 
+export class TableExtractor extends AbstractExtractor{
+    /**
+     *
+     * @param {string} name
+     */
+    constructor(name) {
+        super(name)
+        /** @type {boolean} */
+        this.isTable = true
```

Comment:
Great call!

Reply:
> Instead of `isTable`, a common pattern you'll see is to give each subclass a `name` field. E.g. 
> 
> 
> 
> ```js
> 
> export class TableExtractor extends AbstractExtractor {
> 
>     name = 'table_extractor'
> 
> 
> 
>     constructor(...) { ... }
> 
> }
> 
> ```
> 
> 
> 
> That way we can easily check which extractor is currently running by e.g. `this.extractor.name == 'table_extractor'`